### PR TITLE
Fix travis

### PR DESCRIPTION
**Pull request by aschokking**
Merge upstream fixes from FRCTemplate repo to fix travis ci
@WasabiFan @JohnGilb

### CODE REVIEW

**Comment by aschokking** [comment]
Not sure why this didn't work on travis but I'll need to take another look.

**Comment by aschokking** [comment]
So the problem before was javac puking on lambda since we didn't set java version == 8 explicitly in the seriouslycommonb lib ant files and I guess we weren't using lambda there.

**Comment by aschokking** [comment]
Note: the tests are failing still but the are legitimate test failures (addressed in another branch) as opposed to before where the tests wouldn't run at all (now fixed).

File: CompetitionTemplate/robot-program-build.xml [hunk 1]
```diff
@@ -19,10 +19,12 @@
   <!-- Build JAR for the common lib -->
   <ant antfile="${commonlib.root}/commonlib-build.xml" target="jar-with-test-code" inheritAll="false" />
 
-  <!-- TODO: Do we want this commented out? -->
-  <!-- Original commit that made this change said "Something is wrong with deploying the build on my computer, so disabling copying log config" -->
-  <ant antfile="lib/log4jConfig/deployLogging.xml" target="sendConfig" inheritall="false"/>
-
+  <target name="deploy-logs-and-code" depends="deploy">
+  	<!-- TODO: Do we want this commented out? -->
```
Comment (WasabiFan) [inline CompetitionTemplate/robot-program-build.xml]: We don't need this TODO (or the comment after it)

**Pull request merged**